File: go/valid-parentheses/valid-parentheses.go
package main

import(
	"fmt"
)

func isValid(s string) bool {
    stack := []rune{} // Stack for opening brackets
    hash := map[rune]rune{')': '(', ']': '[', '}': '{'}
    
    for _, char := range s {
        if match, found := hash[char]; found {
            if len(stack) > 0 && stack[len(stack)-1] == match {
                stack = stack[:len(stack)-1]
            } else {
                return false
            }
        } else {
            stack = append(stack, char)
        }
    }
    return len(stack) == 0 
}

func main(){
	input := "()"
	fmt.Println("answer to the first example is:",isValid(input))
}

File: js/valid-parentheses/valid-parentheses.js
function isValid(s){
    let stack = []
    let hash = {
        ")": "(",
        "]": "[",
        "}": "{",
    }
    for (i = 0 ; i < s.length ; i++){
        if (s[i] in hash){
            if(stack.length && stack[stack.length - 1] == hash[s[i]]){
                stack.pop()
            } else {
                return false
            } 
        } else {
            stack.push(s[i])
        }
    }
    return stack.length == 0

}

let s= "()"

console.log("answer to the first example is:",isValid(s))